### PR TITLE
add logrotate for slime log and klog

### PR DESCRIPTION
**Pull request by cywang1905**
Demo files contain slimeboot yaml and pv+pvc yaml and can be seen here https://github.com/slime-io/lazyload/pull/16

### CODE REVIEW

File: boot/helm-charts/slimeboot/templates/deployment.yaml [hunk 1]
```diff
@@ -97,11 +97,17 @@ spec:
           volumeMounts:
             - mountPath: /etc/slime/config
               name: config-volume
+            {{- with $.Values.containers.volumeMounts }}
```
Comment (YonkaFang) [inline boot/helm-charts/slimeboot/templates/deployment.yaml:100]: containers? it looks like an array

File: framework/util/log.go [hunk 2]
```diff
@@ -15,28 +17,39 @@ func TimeEncoder(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
...
-		})
 	}
+	log.SetLevel(level)
+	log.SetFormatter(&log.TextFormatter{
+		TimestampFormat: time.RFC3339,
+	})
 
-	if KlogLevel != 0 {
-		initKlog(KlogLevel)
+	if !logConfig.LogRotate {
```
Comment (YonkaFang) [inline framework/util/log.go:38]:
```go
var (
  output = os.Stdout
  logger *lumberjack.Logger
)

if logConfig.LogRotate {
  logger = &lumberjack.Logger{...}
  output = logger
}

log.SetOutput(os.Stdout)
initKlog(logConfig.KlogLevel, logger)
```

is more clear

File: framework/bootstrap/config.go [hunk 3]
```diff
@@ -79,6 +87,24 @@ func patchGlobal(global, patch *bootconfig.Global) {
 		if global.Log.KlogLevel == 0 {
 			global.Log.KlogLevel = patch.Log.KlogLevel
 		}
+		if global.Log.LogRotate {
```
Comment (YonkaFang) [inline framework/bootstrap/config.go:90]: consider using lib like `github.com/imdario/mergo` to do these `merge-struct` thing to make the code more brief

File: framework/util/log.go (hunk 2 above)
Comment (cywang1905) [reply framework/util/log.go]: done
```go
	var output io.Writer
	output = os.Stdout
	if logConfig.LogRotate {
		output = &lumberjack.Logger{...}
	}
	log.SetOutput(output)
	initKlog(logConfig.KlogLevel, output)

File: boot/helm-charts/slimeboot/templates/deployment.yaml (hunk 1 above)
Comment (cywang1905) [reply boot/helm-charts/slimeboot/templates/deployment.yaml]: chang to $.Values.containers.slime.volumeMounts

**Review by YonkaFang — APPROVED** [review]

**Pull request merged**